spec/rubocop/cop/rspec/subject_declaration_spec.rb

```diff
@@ -0,0 +1,107 @@
+          let!(:subject) { 'some subject' }
+          ^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+
+      it 'is an offense with a block pass' do
+        expect_offense(<<~RUBY)
+          let!(:subject, &block)
+          ^^^^^^^^^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+    end
+
+    context 'with `let` helper' do
+      it 'is an offense without a block pass' do
+        expect_offense(<<~RUBY)
+          let(:subject) { 'some subject' }
+          ^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+
+      it 'is an offense with a block pass' do
+        expect_offense(<<~RUBY)
+          let(:subject, &block)
+          ^^^^^^^^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+    end
+
+    it 'is an offense when declared Ambiguously with `subject`' do
+      expect_offense(<<~RUBY)
+        subject(:subject) { 'some subject' }
+        ^^^^^^^^^^^^^^^^^ Ambiguous declaration of subject
+      RUBY
+    end
+
+    it 'is an offense when declared Ambiguously with `subject!`' do
+      expect_offense(<<~RUBY)
+        subject!(:subject) { 'some subject' }
+        ^^^^^^^^^^^^^^^^^^ Ambiguous declaration of subject
+      RUBY
+    end
+  end
+
+  context 'when `subject` is string' do
+    context 'with `let!` helper' do
+      it 'is an offense without block pass' do
+        expect_offense(<<~RUBY)
+          let!('subject') { 'some subject' }
+          ^^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+
+      it 'is an offense with block pass' do
+        expect_offense(<<~RUBY)
+          let!('subject', &block)
+          ^^^^^^^^^^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+    end
+
+    context 'with `let` helper' do
+      it 'is an offense with a block-pass' do
+        expect_offense(<<~RUBY)
+          let('subject') { 'some subject' }
+          ^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+
+      it 'is an offense with a block pass' do
+        expect_offense(<<~RUBY)
+          let('subject', &block)
+          ^^^^^^^^^^^^^^^^^^^^^^ Use subject explicitly rather than using let
+        RUBY
+      end
+    end
+
+    it 'is an offense when declared ambiguously using `subject`' do
+      expect_offense(<<~RUBY)
+        subject('subject') { 'some subject' }
+        ^^^^^^^^^^^^^^^^^^ Ambiguous declaration of subject
+      RUBY
+    end
+
+    it 'is an offense when declared ambiguously using `subject!`' do
+      expect_offense(<<~RUBY)
+        subject!('subject') { 'some subject' }
+        ^^^^^^^^^^^^^^^^^^^ Ambiguous declaration of subject
+      RUBY
+    end
+  end
+
+  context 'when subject helper is used directly' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        subject(:test_subject) { 'some subject' }
```

Comment:
for spec compliteness, it would be nice to have the rest of the helpers (subject!, let, let!) tested in a situation in which they do not trigger an offense. For subject there is the case with no name at all, e.g. `subject { 'some subject' }`